test/spark/test_0sparkml.py

```diff
@@ -0,0 +1,198 @@
+    automl_experiment = AutoML(task=task)
+    try:
+        duration = automl_experiment.retrain_from_log(
+            X_train=X_train,
+            y_train=y_train,
+            train_full=True,
+            record_id=0,
+            **automl_settings,
+        )
+        print(duration)
+        print(automl_experiment.model)
+        print(automl_experiment.predict(X_train)[:5])
+        print(y_train.to_numpy()[:5])
+    except ValueError:
+        return
+
+
+def test_spark_synapseml():
+    for task in ["classification", "regression", "rank"]:
+        _test_spark_synapseml_lightgbm(spark, task)
+
+
+def test_spark_input_df():
+    df = (
+        spark.read.format("csv")
+        .option("header", True)
+        .option("inferSchema", True)
+        .load(
+            "wasbs://publicwasb@mmlspark.blob.core.windows.net/company_bankruptcy_prediction_data.csv"
+        )
+    )
+    train, test = df.randomSplit([0.8, 0.2], seed=1)
+    feature_cols = df.columns[1:]
+    featurizer = VectorAssembler(inputCols=feature_cols, outputCol="features")
+    train_data = featurizer.transform(train)["Bankrupt?", "features"]
+    test_data = featurizer.transform(test)["Bankrupt?", "features"]
+    automl = AutoML()
+    settings = {
+        "time_budget": 30,  # total running time in seconds
+        "metric": "roc_auc",
+        "estimator_list": [
+            "lgbm_spark"
+        ],  # list of ML learners; we tune lightgbm in this example
+        "task": "classification",  # task type
+        "log_file_name": "flaml_experiment.log",  # flaml log file
+        "seed": 7654321,  # random seed
+    }
+    df = to_pandas_on_spark(to_pandas_on_spark(train_data).to_spark(index_col="index"))
+
+    automl.fit(
+        dataframe=df,
+        label="Bankrupt?",
+        labelCol="Bankrupt?",
+        isUnbalance=True,
+        **settings,
+    )
+
+    try:
+        model = automl.model.estimator
+        predictions = model.transform(test_data)
+
+        from synapse.ml.train import ComputeModelStatistics
+
+        metrics = ComputeModelStatistics(
+            evaluationMetric="classification",
+            labelCol="Bankrupt?",
+            scoredLabelsCol="prediction",
+        ).transform(predictions)
+        metrics.show()
+    except AttributeError:
+        print("No fitted model because of too short training time.")
+
+    # test invalid params
+    settings = {
+        "time_budget": 10,  # total running time in seconds
+        "metric": "roc_auc",
+        "estimator_list": [
+            "lgbm"
+        ],  # list of ML learners; we tune lightgbm in this example
+        "task": "classification",  # task type
+        "use_spark": True,
+    }
+    with pytest.raises(ValueError) as excinfo:
+        automl.fit(
+            dataframe=df,
+            label="Bankrupt?",
+            labelCol="Bankrupt?",
+            isUnbalance=True,
+            **settings,
+        )
+    assert "No estimator is left." in str(excinfo.value)
```

Comment:
This is a good practice to learn. @qingyun-wu @liususan091219 @int-chaos @skzhang1